crates/solvers/src/domain/solution.rs

```diff
@@ -17,9 +17,80 @@ pub struct Solution {
 }
 
 impl Solution {
+    /// Returns `self` with a new score.
     pub fn with_score(self, score: Score) -> Self {
         Self { score, ..self }
     }
+
+    /// Returns `self` with eligible interactions internalized using the
+    /// Settlement contract buffers.
+    ///
+    /// Currently, this internalizes all interactions with input/outputs where
+    /// all input tokens are trusted and the settlement contract has sufficient
+    /// balance to cover the output tokens.
+    pub fn with_buffers_internalizations(mut self, tokens: &auction::Tokens) -> Self {
+        let mut used_buffers = HashMap::new();
+        for interaction in self.interactions.iter_mut() {
+            let (inputs, outputs, internalize) = match interaction {
```

Comment:
do input/outputs need to be mutable?

Reply:
🤦 - no it does not.

Reply:
Fixed in 9899e04c3198d20922d4295cbd2e25a48703a2f4